pkg/compliance/spec/compliance.go

```diff
@@ -70,18 +73,32 @@ func scannerByCheckID(checkID string) types.Scanner {
 	}
 }
 
-// GetComplianceSpec accepct compliance flag name/path and return builtin or file system loaded spec
-func GetComplianceSpec(specNameOrPath string) (ComplianceSpec, error) {
+// GetComplianceSpec accept compliance flag name/path and return builtin or file system loaded spec
+func GetComplianceSpec(specNameOrPath string, c cache.TrivyCache) (ComplianceSpec, error) {
+	if specNameOrPath == "" {
+		return ComplianceSpec{}, nil
+	}
+
 	var b []byte
 	var err error
-	if strings.HasPrefix(specNameOrPath, "@") {
+	if strings.HasPrefix(specNameOrPath, "@") { // load user specified spec from disk
 		b, err = os.ReadFile(strings.TrimPrefix(specNameOrPath, "@"))
 		if err != nil {
-			return ComplianceSpec{}, fmt.Errorf("error retrieving compliance spec from path: %w", err)
+			return ComplianceSpec{}, fmt.Errorf("error retrieving compliance spec from specified path: %w", err)
 		}
 	} else {
-		// TODO: GetSpecByName() should return []byte
-		b = []byte(sp.NewSpecLoader().GetSpecByName(specNameOrPath))
+		_, err := os.ReadFile(filepath.Join(c.GetChecksDir(), "metadata.json"))
```

Comment:
Usually Stat is used to verify the existence of a file

Reply:
Updated